doc/services/storage/nvs/nvs.rst

```diff
@@ -19,11 +19,14 @@ combination of these.
 Each element is stored in flash as metadata (8 byte) and data. The metadata is
 written in a table starting from the end of a nvs sector, the data is
 written one after the other from the start of the sector. The metadata consists
-of: id, data offset in sector, data length, part (unused), and a CRC. The CRC is
+of: id, data offset in sector, data length, part (unused), and a CRC. This CRC is
 only calculated over the metadata and only ensures that a write has been
-completed. The actual data of the element is not protected by a CRC. It is
-encouraged to include a CRC as part of the data and to take appropriate
-corrective actions when the data CRC does not match its expected value.
+completed. The actual data of the element can be protected by a different (and optional)
+CRC-32. Use the :kconfig:option:`CONFIG_NVS_DATA_CRC` configuration item to enable the
+the data part CRC.
+**Note:** the data CRC is checked only when the whole data of the element is read.
```

Comment:
Nit: an extra note could be added to stating that enabling the config option on a nvs system that does not have the crc-32 enabled will make all existing data invalid.

Reply:
I added the note !